src/goto-instrument/contracts/contracts.cpp

```diff
@@ -1346,6 +1365,48 @@ void code_contractst::enforce_contract(const irep_idt &function)
   add_contract_check(original, mangled, wrapper.body);
 }
 
+void code_contractst::assert_function_pointer_obeys_contract(
+  const function_pointer_obeys_contract_exprt &expr,
+  const irep_idt &property_class,
+  const replace_symbolt &replace,
+  const irep_idt &mode,
+  goto_programt &dest)
+{
+  source_locationt loc(expr.source_location());
+  loc.set_property_class(property_class);
+  std::stringstream comment;
+  comment << "Assert function pointer '"
+          << from_expr(ns, "", expr.function_pointer()) << "' obeys contract '"
+          << from_expr(ns, "", expr.contract()) << "'";
```

Comment:
Would any symbol identifier be available that could be used in place of `""`? This is a nit pick, but using `from_expr` with an empty string will mean it falls back to some default language. If we have the symbol name then the language of that symbol's mode will be used.